### DISCUSSION TITLE
[Feat]: Accessibility in the core package

### DISCUSSION THREAD

**Discussion by heresy0-0y** [davidjerleke/embla-carousel#316] [Ideas]
Hi @davidjerleke ! 
It's 2022, accessibility is a requirement for the web. 
It always has been, but especially in recent years, it's become more popularly understood that it increases quality of UX for all users, and significantly affects a sites SEO.

 [https://www.searchenginejournal.com/seo-accessibility/379582/](url)

I would be very surprised if I was in the minority in thinking that accessibility should be part of the core package. There are plenty of standards to look to if you feel you don't know "how accessible" to make embla. WAI, a11y, or even WCAG are a few examples.

I would love to contribute to your library if you were open to something like this!
I saw you mention in past issues that this could be a plugin, so I'm going to start working from that angle, but I'd love your thoughts on considering these features for the core package.

**Comment by davidjerleke** [comment]
Hi @heresy0-0y,

Thanks for your suggestion. Yes, accessibility has been brought up before   and it’s important in 2022. 

> I would be very surprised if I was in the minority in thinking that accessibility should be part of the core package. There are plenty of standards to look to if you feel you don't know "how accessible" to make embla. WAI, a11y, or even WCAG are a few examples.

The Embla Carousel core package is a bare bones library. It solves the hardest technical challenges with building carousels and provides a highly flexible API and plugin system for users to extend it. I have two concerns about adding accessibility directly to the core package:

1. Embla Carousel doesn’t come with previous and next buttons or dot navigation. Autoplay isn’t built-in, it’s a plugin. So adding accessibility to the core package would only include accessibility for the carousel scroll container and slides. But an accessible carousel needs more than that. How would we go about accessibility for previous and next buttons then? Should the accessibility code for these buttons be included in a plugin called `embla-carousel-navigation`? And what about autoplay? Should the accessibility concerns for autoplay be solved in the `embla-carousel-autoplay` package? This approach could get quite messy because the accessibility logic would be sprinkled in different packages. 

2. The core package doesn’t have any knowledge about what the plugins are doing or not. It just initializes the plugins and destroys them when the carousel is destroyed. But the plugins have a lot of information about the Embla instance that initializes them.

This is why I’m leaning towards creating an  `embla-carousel-accessibility` plugin that handles all accessibility instead of sprinkling out the logic.

> I would love to contribute to your library if you were open to something like this!
I saw you mention in past issues that this could be a plugin, so I'm going to start working from that angle, but I'd love your thoughts on considering these features for the core package.

This is great news! Version 7 will be released soon and I think people want an accessibility plugin for Embla. I would be happy to help with my in-depth knowledge of how Embla works internally.

- #298 

Kindly,
David
